pkg/segment/segexecution.go

```diff
@@ -439,6 +439,11 @@ func ExecuteQuery(root *structs.ASTNode, aggs *structs.QueryAggregators, qid uin
 		log.Errorf("qid=%d, ExecuteQuery: failed to get number of RRCs for qid! Error: %v", qid, err)
 	}
 
+	res.RemoteLogs, err = query.GetAllRemoteLogs(res.AllRecords, qid)
```

Comment:
This is fine. 
But, I am not sure whether we should fetch the `RawLogs` here itself, it does not exactly match with our current pattern, where we fetch the raw logs in `searchhandler` or `wsSearchHandler`.

Reply:
This was done to reduce changes in the executeQuery since it is being used at quite a few places and we would need to get remote logs before query gets deleted.